mlrun/execution.py

```diff
@@ -982,7 +989,7 @@ def set_if_not_none(_struct, key, val):
             "metadata.annotations": self._annotations,
             "spec.parameters": self._parameters,
             "spec.outputs": self._outputs,
-            "spec.inputs": {k: v.artifact_url for k, v in self._inputs.items()},
+            "spec.inputs": self._inputs,
```

Comment:
is it backwards compatible? could it be that we still have contexts as objects?

Reply:
the `artifact_url` is now the inputs values so the run object should remain the same.
other than that, I don't see how we can have an old context object with new mlrun code.
if you are in jupyter and upgrading mlrun, restarting the kernel should delete the memory.
if you are in a pod you either have old context object with old mlrun code or new context with new mlrun code because the context is created from the run object during the run flow.
if you are in the server we are ok as well